src/components/TodoInfo/TodoInfo.tsx

```diff
@@ -0,0 +1,131 @@
+import {
+  FC,
+  useState,
+  useEffect,
+  FormEvent,
+  useRef,
+} from 'react';
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todo: Todo,
+  loadingTodosId: number[],
+  onDelete: (todoId: number) => void,
+  onUpdate: (todoId: number, updatedTodo: Partial<Todo>) => void,
+};
+
+export const TodoInfo: FC<Props> = ({
+  todo,
+  loadingTodosId,
+  onDelete,
+  onUpdate,
+}) => {
+  const { title, completed, id } = todo;
+  const [todoTitle, setTodoTitle] = useState(title);
+  const [isEditing, setIsEditing] = useState(false);
+
+  const handleDelete = async (todoId: number) => {
+    await onDelete(todoId);
+  };
+
+  const handleToggle = async (todoId: number) => {
+    await onUpdate(todoId, { completed: !completed });
+  };
+
+  const handleCancel = () => {
+    setIsEditing(false);
+    setTodoTitle(title);
+  };
+
+  const inputField = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (inputField.current) {
+      inputField.current.focus();
+    }
+
+    const handleEscape = (e: KeyboardEvent) => {
+      if (e.key === 'Escape') {
+        handleCancel();
+      }
+    };
+
+    document.addEventListener('keydown', handleEscape);
+
+    return () => {
+      document.removeEventListener('keydown', handleEscape);
+    };
+  }, [isEditing]);
+
+  const handleSubmit = (e: FormEvent) => {
```

Comment:
Consider using useCallback to prevent the unnecessary re-rendering of the component when handleSubmit function is updated. 📝